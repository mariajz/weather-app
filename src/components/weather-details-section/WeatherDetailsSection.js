import React from 'react';
import { CustomImage, CustomIcon } from '../../commons/visual-elements';
import { fetchWeatherImage } from '../../utils/helpers';
import { degreeSymbol } from '../../constants/Constants';
import {
    StyledText as ContentText,
    StyledText as WeatherCondition,
} from '../../commons/styles/CommonStyles';
import {
    LocationWrapper,
    HeadingText,
    SubHeadingText,
    WeatherDetailsWrapper,
    WeatherContentWrapper,
    ItemWrapper,
    WeatherDetailsSectionWrapper,
} from './WeatherDetailsSection.style';

const WeatherDetailsSection = ({
    locationProps,
    weatherIconProps,
    weatherDetailsProps,
    weatherSummaryProps,
}) => {
    const { locationName, region, country } = locationProps;
    const { weatherType } = weatherIconProps;
    const { temperature, condition } = weatherDetailsProps;
    const { wind, uv, humidity } = weatherSummaryProps;

    return (
        <WeatherDetailsSectionWrapper>
            <LocationWrapper>
                <HeadingText>{locationName}</HeadingText>
                <SubHeadingText>
                    {region}, {country}
                </SubHeadingText>
            </LocationWrapper>
            <CustomImage
                path={fetchWeatherImage(weatherType)}
                width={100}
                height={100}
            />
            <WeatherDetailsWrapper>
                <HeadingText>
                    {temperature}
                    {degreeSymbol}C
                </HeadingText>
                <WeatherCondition size={22}>{condition}</WeatherCondition>
            </WeatherDetailsWrapper>
            <WeatherContentWrapper>
                <ItemWrapper>
                    <CustomIcon iconName="wind" />
                    <ContentText size={18} marginLeft={5}>
                        {wind} km
                    </ContentText>
                </ItemWrapper>
                <ItemWrapper>
                    <CustomIcon iconName="humidity" />
                    <ContentText size={18} marginLeft={5}>
                        {humidity}%
                    </ContentText>
                </ItemWrapper>
                <ItemWrapper>
                    <CustomIcon iconName="sun" />
                    <ContentText size={18} marginLeft={5}>
                        {uv} UV
                    </ContentText>
                </ItemWrapper>
            </WeatherContentWrapper>
        </WeatherDetailsSectionWrapper>
    );
};

export default WeatherDetailsSection;
