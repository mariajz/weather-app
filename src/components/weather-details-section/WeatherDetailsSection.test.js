import { render } from '@testing-library/react-native';
import React from 'react';
import WeatherDetailsSection from './WeatherDetailsSection';

const defaultProps = {
    locationProps: {
        country: 'United States of America',
        locationName: 'Manhatten',
        region: 'New York',
    },
    weatherIconProps: {
        weatherType: 'Sunny',
    },
    weatherDetailsProps: {
        condition: 'Sunny',
        temperature: 27.8,
    },
    weatherSummaryProps: {
        humidity: 51,
        uv: 6,
        wind: 3.6,
    },
};
describe('WeatherDetailsSection', () => {
    it('should render WeatherDetailsSection', () => {
        const container = render(<WeatherDetailsSection {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
