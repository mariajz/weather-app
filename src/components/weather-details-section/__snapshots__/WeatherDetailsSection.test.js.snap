// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetailsSection should render WeatherDetailsSection 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "60%",
      "justifyContent": "space-between",
      "paddingBottom": 15,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Manhatten
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 15,
        }
      }
    >
      New York
      , 
      United States of America
    </Text>
  </View>
  <Image
    height={100}
    source={
      {
        "testUri": "../../../assets/images/sun.png",
      }
    }
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
    testID="image"
    width={100}
  />
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      27.8
      Â°
      C
    </Text>
    <Text
      size={22}
      style={
        {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 0,
        }
      }
    >
      Sunny
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/wind.png",
          }
        }
        style={
          {
            "height": 25,
            "width": 25,
          }
        }
        testID="icon"
        tintColor="white"
      />
      <Text
        marginLeft={5}
        size={18}
        style={
          {
            "color": "white",
            "fontSize": 18,
            "marginLeft": 5,
          }
        }
      >
        3.6
         km
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/humidity.png",
          }
        }
        style={
          {
            "height": 25,
            "width": 25,
          }
        }
        testID="icon"
        tintColor="white"
      />
      <Text
        marginLeft={5}
        size={18}
        style={
          {
            "color": "white",
            "fontSize": 18,
            "marginLeft": 5,
          }
        }
      >
        51
        %
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/sun.png",
          }
        }
        style={
          {
            "height": 25,
            "width": 25,
          }
        }
        testID="icon"
        tintColor="white"
      />
      <Text
        marginLeft={5}
        size={18}
        style={
          {
            "color": "white",
            "fontSize": 18,
            "marginLeft": 5,
          }
        }
      >
        6
         UV
      </Text>
    </View>
  </View>
</View>
`;
